### PR TITLE
Fix NPE when ACK grouping tracker checks duplicated message id

### PR DESCRIPTION
**Pull request by BewareMyPower**
### Motivation

Recently I encountered the following NPE but it's hard to reproduce.

```
01:40:14.630 [pulsar-client-io-54-1] WARN  org.apache.pulsar.client.impl.ClientCnx - [10.0.0.44/10.0.0.44:6650] Got exception java.lang.NullPointerException
    at org.apache.pulsar.client.impl.MessageIdImpl.compareTo(MessageIdImpl.java:213)
    at org.apache.pulsar.client.impl.MessageIdImpl.compareTo(MessageIdImpl.java:37)
    at org.apache.pulsar.client.impl.PersistentAcknowledgmentsGroupingTracker.isDuplicate(PersistentAcknowledgmentsGroupingTracker.java:117)
    at org.apache.pulsar.client.impl.ConsumerImpl.messageReceived(ConsumerImpl.java:1013)
```

From the stack we can see NPE was thrown when an ACK grouping tracker checks duplicated message id. The tracker maintains a `LastCumulativeAck` field that has a `MessageIdImpl` type field `messageId`. However, `messageId` could be null after `recycle()` or just was created with a null `MessageIdImpl`. We should check null here. Anyway, it may be introduced from https://github.com/apache/pulsar/pull/8996

### Modifications

- Check null in `PersistentAcknowledgmentsGroupingTracker#isDuplicate` and returns false if it's null, then the tracker will do nothing in `ConsumerImpl#messageReceived`.
- Check null in `MessageIdImpl#compareTo` and throw `UnsupportedOperationException` if the compared object is null.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

**Review by dockerzhang — APPROVED** [review]

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageIdImpl.java [hunk 1]
```diff
@@ -210,7 +210,8 @@ public int compareTo(MessageId o) {
         } else if (o instanceof TopicMessageIdImpl) {
             return compareTo(((TopicMessageIdImpl) o).getInnerMessageId());
         } else {
-            throw new UnsupportedOperationException("Unknown MessageId type: " + o.getClass().getName());
+            final String typeName = (o != null) ? o.getClass().getName() : "null";
```
Comment (eolivelli) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageIdImpl.java:213]: we can add a check in the beginning of the method.

```
if (o == null) {
   throw new UnsupportedOperationException(....)
}
```

**Review by eolivelli — CHANGES_REQUESTED** [review]

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageIdImpl.java (hunk 1 above)
Comment (BewareMyPower) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageIdImpl.java:213]: I agree.

**Comment by BewareMyPower** [comment]
Now all tests passed, PTAL again @eolivelli

**Review by eolivelli — APPROVED** [review]
LGTM

**Pull request merged**